### ISSUE TITLE
How to map to homologous chromosomes?

### ISSUE THREAD

**Issue by ellisztamas** [FelixKrueger/Bismark#625]
Hi Felix,

I am working with F2 individuals from a cross between two accessions of _Arabidopsis thaliana_. We would like to map reads to the de novo assemblies of the two parental accessions rather than the reference genome. At present we are using a concatenated fasta file with the two assemblies plus organelles. 

However, we run into the predictable problem that when reads map similarly well to two homologous chromosomes they are discarded as multimapping. We have tried to mask the regions of the genome for one parent if an individual is known to be homozygous for the other parent's genome, which helps a lot but is only a partial ad hoc fix. It seems that in the long run we will need mapping software which is aware of homologous chromosomes, but we're not there yet.

Since this seems to be where the field is heading with the availability of long-read genome assemblies, we were wondering if you had any thoughts on how to deal with this?

Thanks for any advice you can give!

**Comment by FelixKrueger** [comment]
Hi Tom,

This is a very interesting question, to which I frankly don't have a very good answer (apologies for not replying sooner). I completely agree with you that if it should be more straight forward to align reads in a manner that is aware of homologous chromosomes, especially with long read technologies. I've got a feeling though that this is going to be beyond the scope of Bismark, and will require brighter minds to come up with more appropriate tools...

For your specific situation, you will probably have a mix of several things:
1. regions in the genome that are very much the same between your parental strains
2. regions that carry SNPs between the different strains
3. regions that carry insertion or deletions, or larger structural variants between the parental strains
4. regions that were assembled in only one strain, but are missing in the other some reason

As you say, reads that would align perfectly well to both genome assemblies (situation 1) would be discarded in a default Bismark alignment. I suppose using long read technologies such as Nanopore or Pacbio and e.g. EM-seq with reads stretching sever kb, minimap2 powered alignments could try to keep reads falling into this category to a minimum; however, if your two parental strains are very similar you might still run into this issue.

If the parental strains are very similar, and mostly differ by a defined number of single-nucleotide polymorphisms, a masking strategy such as the one employed in [SNPsplit](https://felixkrueger.github.io/SNPsplit/) might be a worthwhile approach. The advantage we saw in a SNP-only approach to separate our allelic reads is that the coordinate system is the same between your different parental strains. 

As soon as you start changing the genomic coordinate system between genomes or alleles you want to compare with each other (which would be the case for situations 3 and 4), things can get quite messy - and I can't really offer you a solution for this... There might be cool haplotype assembly programs out there that you might want to take a look at (e.g. [WhatsHap](https://whatshap.readthedocs.io/en/latest/)), but I haven't really followed this field myself.

One approach you could maybe pursue in the interim could be to
- step 1: align reads to the concatenated genome you described above, while specifying `--unapped`. Reads that align uniquely to one or the other allele should be placed in this step
- step 2: perform an full genome vs genome alignment of both assemblies so that you know if regions have a corresponding equivalent in the other genome
- step 3 & 4: Align the unmapped fraction from step 1 to genomes of assembly 1 and 2 separately
- step 5: the output of steps 3&4 will need some post-processing to see if the reads align to regions correspond to each  other, equivalent to what is called 'Unassigned' in terms of SNPsplit. I haven't thought this trhough very much, maybe you can also just perform a single alignments to one alllele and then associate the reads to the other allele via the genome-vs-genome table from step 2?

So bottom line, there might be some temporary or as you say ad-hoc fixes for yhe issues at hand, but ultimately I reckon we will need specific tools that are designed to deal with exactly these situations. Apologies if this is rather handwavey....

**Comment by ellisztamas** [comment]
Thanks for the detailed answer!

You have pretty much confirmed what we have been thinking, and in particular that making sure you're really comparing homolgous sites is both essential and tricky. I would only add that it somewhat depends on the question one wants to address, and in our case we will likely have to play that by ear to work out what is possible.

If we come up with something helpful I will post back here in case it is useful for others.

**Comment by FelixKrueger** [comment]
Thanks Tom for the comment. I would indeed appreciate if you could post any relevant updates here, or reach out if there is something I can do to help. Otherwise I'll just go ahead and close the issue for now.
